### ISSUE TITLE
[pyos][docs] Split up API docs to multiple pages

### ISSUE THREAD

**Issue by sneakers-the-rat** [python-graphblas/python-graphblas#409]
Browsing the API docs now, and they're pretty hard to browse because it is all one very long page. This makes it hard for me to get a 'birds eye view' of the package, because I would expect to hit a landing page for API docs that says "here are the top-level modules/types of objects that you might want to use" with a short description of what each type is, and then having a subpage for eg. "Collections" that explains what they are as a group, etc.

This shouldn't be too hard to do, just make an index.rst file for the API docs that then refers to the subpages, just move all the calls to autodoc to separate files.

The API docs are not ~similarly structured to the code, which is fine, but i see that as an additional point of difficulty in maintenance (one would need to know where to put the autodoc calls and insert them manually, rather than self-structuring docs that eg. can just use automodule). I'll raise that as a separate issue once i get a better handle on the code structure

Part of: https://github.com/pyOpenSci/software-submission/issues/81

**Comment by eriknw** [comment]
Just dropping in to say thank you for creating issues such as this one with specific suggestions for improving the documentation and overall user-experience--they're very helpful :)

**Comment by eriknw** [comment]
I agree with this. For example, the API docs for Matrix, Vector, and Scalar should be on different pages instead of on one long page, right?
